### ISSUE TITLE
Unloading GGML models from UI doesn't work at all. 

### ISSUE THREAD

**Issue by cow-diffusion** [oobabooga/text-generation-webui#2385]
- Clicking 'Unload the model' does nothing when a GGML model is loaded. 
- Loading another model will not unload the loaded GGML model either. 

Seems like the only way to get your VRAM back is to terminate the whole instance and reload (which is super frustrating because that means there is no way to change gpu layer offloading on the fly or even load a different model without an enviable crash) 

EDIT: I discovered I'm an idiot and I forgot I manually updated to llama.cpp to 1.57 to even use GPU offloading for GGML models to begin with. I suppose it would be unreasonable to suspect the web UI would work appropriately when using a non-default config. That being said, I do hope there are plans to update llama.cpp for native GPU offloading support.

**Comment by matatonic** [comment]
My experience as well

**Comment by Daeformator** [comment]
Are you able to actually get the GGML to offload to GPU, how so? (CUDA?)

**Comment by matatonic** [comment]
@Daeformator Yes, the setting is n_gpu_layers, set that to more than the layers (ex. 60 for 30B models) and it will GPU offload. you need the latest of essentially everything.

**Comment by Kwisss** [comment]
@Daeformator check here: https://github.com/oobabooga/text-generation-webui/blob/main/docs/llama.cpp-models.md#gpu-offloading
I think you do need Microsoft build tools installed, but nothing you can't find out with google and the error outputs.

**Comment by TFWol** [comment]
@Kwisss Is this where the error `'LlamaCppModel' object has no attribute 'model'` occurs if attempting to load a ggml model with the llama-cpp-python that's installed by default with text-gen?

**Comment by cow-diffusion** [comment]
> Are you able to actually get the GGML to offload to GPU, how so? (CUDA?)

Please see:
https://github.com/oobabooga/text-generation-webui/blob/main/docs/llama.cpp-models.md

**Comment by TFWol** [comment]
@cow-diffusion Kwisss already [got it covered](https://github.com/oobabooga/text-generation-webui/issues/2385#issuecomment-1566117574)
